activejob/lib/active_job/exceptions.rb

```diff
@@ -78,10 +88,18 @@ def retry_on(*exceptions, wait: 3.seconds, attempts: 5, queue: nil, priority: ni
       #  end
       def discard_on(*exceptions)
         rescue_from(*exceptions) do |error|
-          if block_given?
-            yield self, error
-          else
-            logger.error "Discarded #{self.class} due to a #{error.class}. The original exception was #{error.cause.inspect}."
+          payload = {
+            job: self,
+            adapter: self.class.queue_adapter,
+            error: error,
```

Comment:
I think our style (although there's no rubocop on for it yet) is to not have a trailing comma. Or maybe I imagined that because I personally don't like them 😉